diff --git a/rootdir/etc/init.goldfish.rc b/rootdir/etc/init.goldfish.rc
index 83b7f8a..28d15f7 100644
--- a/rootdir/etc/init.goldfish.rc
+++ b/rootdir/etc/init.goldfish.rc
@@ -13,6 +13,12 @@ on boot
     setprop ro.build.product generic
     setprop ro.product.device generic
 
+# Bluetooth related settings
+    chown bluetooth bluetooth /sys/class/rfkill/rfkill0/state
+    chmod 0660 /sys/class/rfkill/rfkill0/state
+    chown bluetooth bluetooth /dev/ttyS2
+    chmod 0660 /dev/ttyS2
+
 # fake some battery state
     setprop status.battery.state Slow
     setprop status.battery.level 5
@@ -76,3 +82,10 @@ service qemud /system/bin/qemud
 
 service goldfish-logcat /system/bin/logcat -Q
     oneshot
+
+service hciattach /system/bin/sh /system/etc/init.goldfish.bt.sh
+    class late_start
+    user bluetooth
+    group bluetooth net_bt_admin
+    disabled
+    oneshot
